bokeh/core/properties.py

```diff
@@ -1181,7 +1187,9 @@ def _sphinx_type(self):
     def from_json(self, json, models=None):
         if json is None:
             return None
-        elif isinstance(json, list):
+        elif isinstance(json, bokeh_seq_types):
+            if isinstance(self.item_type, Any):
+                return json
```

Comment:
I'm confused about what's happening here; I haven't read the whole PR yet as I'm reading this, but here are the questions I have looking at these lines:

 * how does json have a numpy array in it now when we didn't used to handle that (shouldn't json be plain json types?)
 * Any.from_json is now only called for single Any values but not for arrays, which could be dangerous
 * what happens to existing code that has a `Seq(Any)` where the Any actually does need from_json to be called; it could be any type at all in there, including objects, as far as I remember?